export/cloud_datastore_admin.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Request handler for exporting backups."""
+
+import datetime
+import httplib
+import json
+import logging
+
+from google.appengine.api import app_identity
+from google.appengine.api import urlfetch
+import webapp2
+
+
+class Export(webapp2.RequestHandler):
+
+    def get(self):
+        access_token, _ = app_identity.get_access_token(
+            'https://www.googleapis.com/auth/datastore')
+        app_id = app_identity.get_application_id()
+        timestamp = datetime.datetime.utcnow().strftime('%Y%m%d-%H%M%S')
+
+        output_url_prefix = self.request.get('output_url_prefix')
+        assert output_url_prefix and output_url_prefix.startswith('gs://')
+        if '/' not in output_url_prefix[5:]:
+             # Only a bucket name has been provided - no prefix or trailing
+             # slash.
+            output_url_prefix += '/' + timestamp
+        else:
+            output_url_prefix += timestamp
+
+        entity_filter = {
+            'kinds': self.request.get_all('kind'),
+            'namespace_ids': self.request.get_all('namespace_id')
+        }
+        request = {
+            'project_id': app_id,
+            'output_url_prefix': output_url_prefix,
+            'entity_filter': entity_filter
+        }
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': 'Bearer ' + access_token
+        }
+        url = 'https://datastore.googleapis.com/v1/projects/%s:export' % app_id
+        try:
+            result = urlfetch.fetch(
+                url=url,
+                payload=json.dumps(request),
+                method=urlfetch.POST,
+                deadline=60,
+                headers=headers)
+            if result.status_code == httplib.OK:
+                logging.info(result.content)
```

Comment:
That's going to be lots of logs, maybe log a summary at the end of how many data items were successfully exported instead.

Reply:
How can you tell this is going to be a lot of logs?

Reply:
Wouldn't it just be one log then, from one request?

Reply:
Sorry, you're right! I am not sure what I was thinking here, for some reason I thought there was a "for" loop. Please ignore this comment.